apt-transport-https

hex2a opened the pull request:
declaring the apt-transport-https package may cause a duplicate resource declaration:

`Error: Evaluation Error: Error while evaluating a Function Call, Duplicate declaration: Package[apt-transport-https] is already declared`

there is a dependency cycle when trying to order apt-transport-https before apt_update (as keys/sources are ordered before apt_update which is ordered before apt install for new packages, causing apt_update to fail since the https source is already added). 

Since the $required_packages parameter in the apt module is deprecated, this package should be already installed in the system.


Pull request acceptance prerequisites:

- [X] Signed the [CLA](https://www.elastic.co/contributor-agreement/) (if not already signed)
- [X] Rebased/up-to-date with base branch
- [X] Updated CHANGELOG.md with patch notes (if necessary)
- [X] Any relevant docs (README.markdown or inline documentation) updated (if necessary)
- [X] Updated CONTRIBUTORS (if you would like attribution)
- [ ] Tests pass CI

tylerjl commented:
Thanks for the PR @hex2a!

While the acceptance test failures are broken tests I need to fix, [some of the rspec failures are legitimate that need to have updated tests](https://travis-ci.org/elastic/puppet-kibana/jobs/211737530#L1142). You can test these resources with

    $ rspec spec/classes/kibana_spec.rb

hex2a commented:
I modified the pull request, and removed the apt-transport-https from puppet manifests/tests, the package is now being installed for the test environment only, similar to the elastic/elasticsearch and elastic/logstash puppet modules.

tylerjl commented:
Sounds good. I added the package resource as a convenience given that users may not be managing the package, but we can revisit it if users encounter problems with the repos being served over HTTPS. Thanks!
